src/middlewared/middlewared/plugins/disk.py

```diff
@@ -143,9 +142,9 @@ async def disk_extend(self, disk, context):
             if await self.middleware.call('truenas.is_ix_hardware'):
                 disk['supports_smart'] = True
             else:
-                disk['supports_smart'] = disk['name'].startswith('nvme') or bool(RE_SMART_AVAILABLE.search(
-                    await self.middleware.call('disk.smartctl', disk['name'], ['-a'], {'silent': True}) or ''
-                ))
+                disk_query = await self.middleware.call('disk.smartctl', disk['name'], ['-a', '--json=c'], {'silent': True})
```

Comment:
This is performed even if `disk['name'].startswith('nvme')` is `True`, which is unnecessary.